Source/CesiumRuntime/Public/CesiumMetadataPrimitive.h

```diff
@@ -49,23 +45,18 @@ struct CESIUMRUNTIME_API FCesiumMetadataPrimitive {
    * extension
    * @param metadata The EXT_feature_metadata of the gltf mesh primitive.
    * primitive
+   * @param primitiveFeatures The FCesiumPrimitiveFeatures denoting the feature
+   * IDs in the glTF mesh primitive.
    */
   FCesiumMetadataPrimitive(
       const CesiumGltf::Model& model,
       const CesiumGltf::MeshPrimitive& primitive,
       const CesiumGltf::ExtensionMeshPrimitiveExtFeatureMetadata& metadata,
-      const CesiumGltf::ExtensionModelExtFeatureMetadata& modelMetadata);
+      const FCesiumPrimitiveFeatures& primitiveFeatures);
 
 private:
-  TArray<FCesiumFeatureIdAttribute> _featureIdAttributes;
-  TArray<FCesiumFeatureIdTexture> _featureIdTextures;
+  const FCesiumPrimitiveFeatures* _pPrimitiveFeatures;
```

Comment:
Looking closer, maybe it's ok because instances of both `FCesiumPrimitiveFeatures` and `FCesiumMetadataPrimitive` are stored side-by-side in `UCesiumGltfPrimitiveComponent`, so they have the same lifetime.

If that will always be the case, you can ignore my suggestion above. But it could be worth a note in the doc for the constructor about this, because it violates C++ norms a bit.

Reply:
Yeah, this relationship is awkward. Ideally `FCesiumMetadataPrimitive` wouldn't have to exist (it will be replaced by a new metadata implementation). But to keep backwards compatibility, it needs to be aware of the new implementations and use _their_ functions in order to maintain its functionality. Hence the side-by-side construction and pointers.

I would prefer to keep `FCesiumPrimitiveFeatures` independent of `FCesiumMetadataPrimitive`. The `EXT_mesh_features` extension can exist independently of `EXT_structural_metadata` and I think the implementation should reflect that. So I'll go with the second suggestion, but If there's a way to better do this while adhering to C++ norms, let me know.